### PR TITLE
[Supplier] Refactor/non custodial staking tests

### PR DESCRIPTION
**Pull request by red-0ne**
## Summary

This PR is a follow-up to #716.
* It refactors the existing unit/integration tests to add `Supplier.OwnerAddress`
* Adapts the `Supplier` (un)staking and tests (staking config, (un)staking CLI and keeper methods) to comply with the `Supplier` non-custodial staking.

## Issue

- #493 

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

**Documentation changes** (only if making doc changes)
- [ ] `make docusaurus_start`; only needed if you make doc changes

**Local Testing** (only if making code changes)
- [x] **Unit Tests**: `make go_develop_and_test`
- [ ] **LocalNet E2E Tests**: `make test_e2e`
- See [quickstart guide](https://dev.poktroll.com/developer_guide/quickstart) for instructions

**PR Testing** (only if making code changes)
- [ ] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.
    - **THIS IS VERY EXPENSIVE**, so only do it after all the reviews are complete.
    - Optionally run `make trigger_ci` if you want to re-trigger tests without any code changes
    - If tests fail, try re-running failed tests only using the GitHub UI as shown [here](https://github.com/pokt-network/poktroll/assets/1892194/607984e9-0615-4569-9452-4c730190c1d2)

## Sanity Checklist

- [ ] I have tested my changes using the available tooling
- [ ] I have commented my code
- [ ] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

## Summary by CodeRabbit

- **New Features**
	- Enhanced readability of output messages in the Makefile.
	- Added `OwnerAddress` to the `Supplier` data model for better ownership tracking.
	- Introduced support for both owner and operator addresses in various staking and unstaking processes.

- **Bug Fixes**
	- Updated tests for `MsgUnstakeSupplier` to correctly validate signer addresses and enhance robustness.

- **Documentation**
	- Improved test coverage and clarity for supplier-related functionality, reflecting the new data model changes.

### CODE REVIEW

File: x/supplier/types/message_stake_supplier_test.go [hunk 1]
```diff
@@ -36,61 +41,141 @@ func TestMsgStakeSupplier_ValidateBasic(t *testing.T) {
...
+				Stake:        &sdk.Coin{Denom: volatile.DenomuPOKT, Amount: math.NewInt(100)},
+				Services:     defaultServicesList,
+			},
+			expectedErr: ErrSupplierInvalidAddress,
+		},
+		{
+			desc: "missing owner address",
+			msg: MsgStakeSupplier{
+				Sender:   ownerAddress,
+				Address:  operatorAddress,
```
Comment (Olshansk) [inline x/supplier/types/message_stake_supplier_test.go:99]: Can you add a commented out line with the key, it'll be easier to see.

File: x/supplier/types/message_stake_supplier_test.go [hunk 2]
```diff
@@ -36,61 +41,141 @@ func TestMsgStakeSupplier_ValidateBasic(t *testing.T) {
...
+				Stake:    &sdk.Coin{Denom: volatile.DenomuPOKT, Amount: math.NewInt(100)},
 				Services: defaultServicesList,
 			},
 			expectedErr: ErrSupplierInvalidAddress,
 		},
+		{
+			desc: "missing operator address",
+			msg: MsgStakeSupplier{
+				Sender:       ownerAddress,
+				OwnerAddress: ownerAddress,
```
Comment (Olshansk) [inline x/supplier/types/message_stake_supplier_test.go:109]: Can you add a commented out line with the key, it'll be easier to see.

File: x/supplier/types/message_stake_supplier_test.go [hunk 3]
```diff
@@ -36,61 +41,141 @@ func TestMsgStakeSupplier_ValidateBasic(t *testing.T) {
...
+				Sender:       ownerAddress,
+				OwnerAddress: "invalid_address",
+				Address:      operatorAddress,
+				Stake:        &sdk.Coin{Denom: volatile.DenomuPOKT, Amount: math.NewInt(100)},
+				Services:     defaultServicesList,
+			},
+			expectedErr: ErrSupplierInvalidAddress,
+		},
+		{
+			desc: "missing owner address",
```
Comment (Olshansk) [inline x/supplier/types/message_stake_supplier_test.go:96]: Do you have one with a missing signer (i.e. sender)?

File: x/supplier/types/message_stake_supplier_test.go [hunk 4]
```diff
@@ -36,61 +41,141 @@ func TestMsgStakeSupplier_ValidateBasic(t *testing.T) {
...
-				// Stake explicitly omitted
+				Sender:       ownerAddress,
+				OwnerAddress: ownerAddress,
+				Address:      ownerAddress,
+				Stake:        &sdk.Coin{Denom: volatile.DenomuPOKT, Amount: math.NewInt(100)},
+				Services:     defaultServicesList,
+			},
+		},
+		{
+			desc: "valid different owner and operator address",
```
Comment (Olshansk) [inline x/supplier/types/message_stake_supplier_test.go:54]: Do you have one where the sender/signer is neither the operator or owner?

File: x/supplier/module/tx_unstake_supplier.go [hunk 5]
```diff
@@ -17,7 +17,7 @@ func CmdUnstakeSupplier() *cobra.Command {
 		Long: `Unstake an supplier with the provided parameters. This is a broadcast operation that will unstake the supplier specified by the <operator_address> and owned by 'from' address.
 
 Example:
-$ poktrolld tx supplier unstake-supplier $(OPERATOR_ADDRESS) --keyring-backend test --from $(OWNER_ADDRESS) --node $(POCKET_NODE) --home=$(POKTROLLD_HOME)`,
+$ poktrolld tx supplier unstake-supplier $(OPERATOR_ADDRESS) --keyring-backend test --from $(OWNER_ADDRESS) --node $(POCKET_NODE) --home $(POKTROLLD_HOME)`,
```
Comment (Olshansk) [inline x/supplier/module/tx_unstake_supplier.go:20]: --from can also be the oeprator. Please double check with morse.

File: x/supplier/config/supplier_configs_reader_test.go [hunk 6]
```diff
@@ -360,15 +484,64 @@ func Test_ParseSupplierConfigs_Services(t *testing.T) {
...
+				    endpoints:
+				    - publicly_exposed_url: http://pokt.network:8081
+				      rpc_type: json_rpc
+				      config:
+				        timeout: 10
+				`, operatorAddress),
+			expectedError: config.ErrSupplierConfigInvalidAddress,
+		},
+		{
+			desc: "invalid operator address",
```
Comment (Olshansk) [inline x/supplier/config/supplier_configs_reader_test.go:523]: Is there one when operator_address is explicitly ommited for a success scenario?

**Review by Olshansk — CHANGES_REQUESTED** [review]
Left another roudn of comments. Will be more detailed in the next one.

File: x/supplier/types/message_stake_supplier_test.go (hunk 4 above)
Comment (red-0ne) [reply x/supplier/types/message_stake_supplier_test.go:54]: Good catch!

File: x/supplier/config/supplier_configs_reader_test.go (hunk 6 above)
Comment (red-0ne) [reply x/supplier/config/supplier_configs_reader_test.go:523]: Added

File: x/supplier/module/tx_unstake_supplier_test.go [hunk 7]
```diff
@@ -25,11 +25,14 @@ func TestCLI_UnstakeSupplier(t *testing.T) {
...
+	accounts := testutil.CreateKeyringAccounts(t, kr, 2)
+	operatorAccount := accounts[0]
+	ownerAccount := accounts[1]
 
-	// Initialize the Supplier Account by sending it some funds from the validator account that is part of genesis
-	network.InitAccount(t, net, supplierAccount.Address)
+	// Initialize the Supplier operator account by sending it some funds from the
+	// validator account that is part of genesis
+	network.InitAccountWithSequence(t, net, operatorAccount.Address, 1)
+	network.InitAccountWithSequence(t, net, ownerAccount.Address, 2)
```
Comment (Olshansk) [inline x/supplier/module/tx_unstake_supplier_test.go:35]: Really need to remember this helper, it's great.

File: x/supplier/module/tx_unstake_supplier_test.go [hunk 8]
```diff
@@ -42,24 +45,45 @@ func TestCLI_UnstakeSupplier(t *testing.T) {
...
-		expectedErr *sdkerrors.Error
+		desc            string
+		signerAddress   string
+		operatorAddress string
+		expectedErr     *sdkerrors.Error
 	}{
 		{
-			desc:    "unstake supplier: valid",
-			address: supplierAccount.Address.String(),
+			desc:            "owner unstake supplier: valid",
```
Comment (Olshansk) [inline x/supplier/module/tx_unstake_supplier_test.go:54]: Nice tests

File: x/supplier/keeper/msg_server_unstake_supplier_test.go [hunk 9]
```diff
@@ -182,11 +194,42 @@ func TestMsgServer_UnstakeSupplier_FailIfCurrentlyUnstaking(t *testing.T) {
 	require.ErrorIs(t, err, types.ErrSupplierIsUnstaking)
 }
 
+func TestMsgServer_UnstakeSupplier_OperatorCanUnstake(t *testing.T) {
+	supplierModuleKeepers, ctx := keepertest.SupplierKeeper(t)
+	srv := keeper.NewMsgServerImpl(*supplierModuleKeepers.Keeper)
+
+	// Generate an address for the supplier
+	ownerAddr := sample.AccAddress()
+	supplierAddr := sample.AccAddress()
```
Comment (Olshansk) [inline x/supplier/keeper/msg_server_unstake_supplier_test.go:203]: This is renamed to operator in the next one?

File: x/supplier/keeper/msg_server_unstake_supplier_test.go [hunk 10]
```diff
@@ -182,11 +194,42 @@ func TestMsgServer_UnstakeSupplier_FailIfCurrentlyUnstaking(t *testing.T) {
...
+		Signer:  supplierAddr,
+		Address: supplierAddr,
+	}
+	_, err = srv.UnstakeSupplier(ctx, unstakeMsg)
+	require.NoError(t, err)
+
+	// Make sure the supplier entered the unbonding period
+	foundSupplier, isSupplierFound := supplierModuleKeepers.GetSupplier(ctx, supplierAddr)
+	require.True(t, isSupplierFound)
+	require.True(t, foundSupplier.IsUnbonding())
```
Comment (Olshansk) [inline x/supplier/keeper/msg_server_unstake_supplier_test.go:223]: Do we have a unit test to see where the funds go once unbonding finishes?

Let's add a TODO if not.

File: x/supplier/keeper/msg_server_stake_supplier_test.go [hunk 11]
```diff
@@ -155,22 +148,124 @@ func TestMsgServer_StakeSupplier_FailWithNonExistingService(t *testing.T) {
...
+	require.NoError(t, err)
+
+	// Check that the supplier was created
+	newSupplier, newSupplierFound := k.GetSupplier(ctx, stakeMsg.Address)
+	require.True(t, newSupplierFound)
+
+	oldSupplier, oldSupplierFound := k.GetSupplier(ctx, operatorAddr)
+	require.True(t, oldSupplierFound)
+
+	require.NotEqual(t, oldSupplier.Address, newSupplier.Address)
```
Comment (Olshansk) [inline x/supplier/keeper/msg_server_stake_supplier_test.go:221]:
```suggestion
	// Ensure the operator address was updated.
	require.NotEqual(t, oldSupplier.Address, newSupplier.Address)
```

File: x/supplier/keeper/msg_server_stake_supplier_test.go [hunk 12]
```diff
@@ -155,22 +148,124 @@ func TestMsgServer_StakeSupplier_FailWithNonExistingService(t *testing.T) {
...
+
+	// Trying to update the supplier's owner address using the owner as a signer
+	// will create a new supplier
+	stakeMsg = stakeSupplierForServicesMsg(ownerAddr, operatorAddr, 400, "svcId")
+	stakeMsg.Address = sample.AccAddress()
```
Comment (Olshansk) [inline x/supplier/keeper/msg_server_stake_supplier_test.go:226]: 1. Ditto regarding the clarity of what the stakeMsg is.
2. Shouldn't this be setting `OwnerAddress` and not `Address`?

Confusing w.r.t the comment above.

File: x/supplier/keeper/msg_server_stake_supplier_test.go [hunk 13]
```diff
@@ -155,22 +148,124 @@ func TestMsgServer_StakeSupplier_FailWithNonExistingService(t *testing.T) {
...
+	stakeMsg = stakeSupplierForServicesMsg(ownerAddr, operatorAddr, 400, "svcId")
+	stakeMsg.OwnerAddress = newOwnerAddress
+	_, err = srv.StakeSupplier(ctx, stakeMsg)
+	require.NoError(t, err)
+
+	// Check that the supplier was updated.
+	foundSupplier, supplierFound = k.GetSupplier(ctx, operatorAddr)
+	require.True(t, supplierFound)
+	require.Equal(t, newOwnerAddress, foundSupplier.OwnerAddress)
+}
```
Comment (Olshansk) [inline x/supplier/keeper/msg_server_stake_supplier_test.go:257]: @red-0ne I will re0review this test after the changes.

File: x/supplier/keeper/msg_server_unstake_supplier_test.go (hunk 9 above)
Comment (red-0ne) [reply x/supplier/keeper/msg_server_unstake_supplier_test.go:203]: Yes

File: x/supplier/keeper/msg_server_unstake_supplier_test.go (hunk 10 above)
Comment (red-0ne) [reply x/supplier/keeper/msg_server_unstake_supplier_test.go:223]: Good catch!

File: x/supplier/keeper/msg_server_stake_supplier_test.go (hunk 11 above)
Comment (red-0ne) [reply x/supplier/keeper/msg_server_stake_supplier_test.go:221]: Actually, the operator address is not updated but a new supplier gets created.

Will add a comment.

File: x/supplier/keeper/msg_server_stake_supplier_test.go [hunk 14]
```diff
@@ -155,22 +148,128 @@ func TestMsgServer_StakeSupplier_FailWithNonExistingService(t *testing.T) {
...
+	setStakeMsgSigner(stakeMsgUpdateUrl, operatorAddr)
+	_, err = srv.StakeSupplier(ctx, stakeMsgUpdateUrl)
+	require.NoError(t, err)
+
+	// Check that the supplier was updated
+	foundSupplier, supplierFound := k.GetSupplier(ctx, operatorAddr)
+	require.True(t, supplierFound)
+	require.Equal(t, operatorUpdatedServiceUrl, foundSupplier.Services[0].Endpoints[0].Url)
+
+	// Update the supplier using the owner address as the singer and verify that it succeeds.
```
Comment (Olshansk) [inline x/supplier/keeper/msg_server_stake_supplier_test.go:194]:
```suggestion
	// Update the supplier URL by using the owner address as the singer and verify that it succeeds.
```

File: x/supplier/keeper/msg_server_stake_supplier_test.go [hunk 15]
```diff
@@ -155,22 +148,128 @@ func TestMsgServer_StakeSupplier_FailWithNonExistingService(t *testing.T) {
...
+	stakeMsgUpdateUrl.Services[0].Endpoints[0].Url = ownerUpdaterServiceUrl
+	stakeMsgUpdateUrl.Stake.Amount = math.NewInt(300)
+	setStakeMsgSigner(stakeMsgUpdateUrl, ownerAddr)
+	_, err = srv.StakeSupplier(ctx, stakeMsgUpdateUrl)
+	require.NoError(t, err)
+
+	// Check that the supplier was updated
+	foundSupplier, supplierFound = k.GetSupplier(ctx, operatorAddr)
+	require.True(t, supplierFound)
+	require.Equal(t, ownerUpdaterServiceUrl, foundSupplier.Services[0].Endpoints[0].Url)
```
Comment (Olshansk) [inline x/supplier/keeper/msg_server_stake_supplier_test.go:205]: Can you add a not equal w.r.t the old URL just for sanity?

**Review by Olshansk — APPROVED** [review]
Just an OPTIONAL minor NIT, but otherwise looks g2g!

**Pull request merged**